doc/releases/changelog-dev.md

```diff
@@ -397,6 +397,10 @@
 
 <h3>Bug fixes</h3>
 
+* Pytest now ignores any `DeprecationWarning` raised within autograd's `numpy_wrapper` module.
+  Other assorted minor test warnings are fixed.
+  [(#2007)](https://github.com/PennyLaneAI/pennylane/pull/2007)
+
```

Comment:
Not sure if this is a bug fix per se, perhaps it's an improvement rather. Testing is not user-facing, whereas bug fixing would entail user-facing behaviour change. How about we move it to the Improvements section?

Reply:
I consider some of the warning fixes a "bug fix", since we are removing bad behaviour.

Reply:
@albi3ro I'm not sure I agree here. Using `pytest` is not user-facing, so this entry, as-is might confuse users: they don't use `pytest` with PennyLane, then what exactly changed here?

For those cases that are user-facing, we could have another entry, but those will not be bug fixes either: the code was working, although features to be outdated were used, hence deprecation warnings were emitted. They would likely more so be improvements.

Reply:
Yes, I agree with this being more of an improvement rather than a bug fix